From 313a7fe0f9479cdb815a1f4e474b2b83108c4044 Mon Sep 17 00:00:00 2001
From: Shuguang Gong <shuguang.gong@intel.com>
Date: Fri, 27 Mar 2015 18:21:29 +0800
Subject: [PATCH 338/338] ov2685: add new resolution settings

Current CIF/720P are cropped and downscaled by ISP
from full size of sensor output. New settings are more
accurate and can decrease payload of ISP

Change-Id: I9dfc1503923a54a66c438212f733ea164cd0cd6c
Tracked-On: https://hsdes.intel.com/home/default.html#article?id=1503940663
Signed-off-by: Shuguang Gong <shuguang.gong@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/48486
Approver: Jing Tao <jing.tao@intel.com>
Integrator: Wei Tao <wei.tao@intel.com>
Build: IMIN-legacy (SC)
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../camera/drivers/media/i2c/ov2685.c              |  115 +-
 .../camera/drivers/media/i2c/ov2685.h              | 1274 ++++++++++++++++++--
 2 files changed, 1241 insertions(+), 148 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov2685.c b/drivers/external_drivers/camera/drivers/media/i2c/ov2685.c
index 00abc03..26241d7 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov2685.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov2685.c
@@ -989,54 +989,74 @@ static int ov2685_s_power(struct v4l2_subdev *sd, int power)
 #define OV2585_3_2 1500 /* 3:2*//* 720x480*/
 #define OV2685_16_9 1777 /* 16:9*//* 1280x720*/
 
-static int ov2685_try_res(u32 *w, u32 *h)
-{
-	int i;
-	/*
-	 * The mode list is in ascending order. We're done as soon as
-	 * we have found the first equal or bigger size.
-	 */
-	for (i = 0; i < N_RES; i++) {
-		if (ov2685_res[i].width >= *w &&
-		    ov2685_res[i].height >= *h)
-			break;
-	}
 
-	/*
-	 * If no mode was found, it means we can provide only a smaller size.
-	 * Returning the biggest one available in this case.
-	 */
-	if (i == N_RES)
-		i--;
-
-	*w = ov2685_res[i].width;
-	*h = ov2685_res[i].height;
-	return 0;
+#define LARGEST_ALLOWED_RATIO_MISMATCH 600
+static int distance(struct ov2685_res_struct const *res, u32 w, u32 h)
+{
+	unsigned int w_ratio;
+	unsigned int h_ratio;
+	int match;
+	unsigned int allowed_ratio_mismatch = LARGEST_ALLOWED_RATIO_MISMATCH;
+
+	if (w == 0)
+		return -1;
+	w_ratio = (res->width << 13) / w;
+	if (h == 0)
+		return -1;
+	h_ratio = (res->height << 13) / h;
+	if (h_ratio == 0)
+		return -1;
+	match   = abs(((w_ratio << 13) / h_ratio) - ((int)8192));
+
+	if ((w_ratio < (int)8192) || (h_ratio < (int)8192)  ||
+		(match > allowed_ratio_mismatch))
+		return -1;
+
+	return w_ratio + h_ratio;
 }
 
-static int ov2685_to_res(u32 w, u32 h)
+/* Return the nearest higher resolution index */
+static int nearest_resolution_index(struct v4l2_subdev *sd, u32 w, u32 h)
 {
-	int  index;
+	int i;
+	int idx = -1;
+	int dist;
+	int min_dist = INT_MAX;
+	const struct ov2685_res_struct *tmp_res = NULL;
 
-	for (index = 0; index < N_RES; index++) {
-		if (ov2685_res[index].width == w &&
-		    ov2685_res[index].height == h)
-			break;
+	for (i = 0; i < N_RES; i++) {
+		tmp_res = &ov2685_res[i];
+		dist = distance(tmp_res, w, h);
+		if (dist == -1)
+			continue;
+		if (dist < min_dist) {
+			min_dist = dist;
+			idx = i;
+		}
 	}
-
-	/* No mode found */
-	if (index >= N_RES)
-		index--;
-
-	return index;
+	return idx;
 }
 
 static int ov2685_try_mbus_fmt(struct v4l2_subdev *sd,
 				struct v4l2_mbus_framefmt *fmt)
 {
-	int ret = ov2685_try_res(&fmt->width, &fmt->height);
+	struct ov2685_device *dev = to_ov2685_sensor(sd);
+	int idx;
+
+	mutex_lock(&dev->input_lock);
+
+	idx = nearest_resolution_index(sd, fmt->width, fmt->height);
+
+	if (idx == -1)
+		idx = N_RES - 1;
+
+	fmt->width = ov2685_res[idx].width;
+	fmt->height = ov2685_res[idx].height;
 	fmt->code = V4L2_MBUS_FMT_UYVY8_1X16;
-	return ret;
+
+	mutex_unlock(&dev->input_lock);
+
+	return 0;
 }
 
 static int ov2685_g_mbus_fmt(struct v4l2_subdev *sd,
@@ -1072,17 +1092,20 @@ static int ov2685_s_mbus_fmt(struct v4l2_subdev *sd,
 	}
 
 	mutex_lock(&dev->input_lock);
-	ov2685_try_res(&width, &height);
-	res_index = ov2685_to_res(width, height);
+
+	res_index = nearest_resolution_index(sd, width, height);
+	if (res_index == -1)
+		res_index = N_RES - 1;
 
 	switch (ov2685_res[res_index].res)  {
-	/*
-	case OV2685_RES_VGA:
-		ret = ov2685_write_reg_array(c, ov2685_vga_init);
+	case OV2685_RES_XCIF:
+		ret = ov2685_write_reg_array(client, ov2685_732_600_init);
+		break;
+	case OV2685_RES_SVGA:
+		ret = ov2685_write_reg_array(client, ov2685_svga_init);
 		break;
-	*/
-	case OV2685_RES_720P:
-		ret = ov2685_write_reg_array(client, ov2685_720p_init);
+	case OV2685_RES_1M3:
+		ret = ov2685_write_reg_array(client, ov2685_1M3_init);
 		break;
 	case OV2685_RES_2M:
 		ret = ov2685_write_reg_array(client, ov2685_2M_init);
@@ -1107,8 +1130,8 @@ static int ov2685_s_mbus_fmt(struct v4l2_subdev *sd,
 	 * because it does not happen with streaming disabled.
 	 */
 	dev->fmt_idx = res_index;
-	fmt->width = width;
-	fmt->height = height;
+	fmt->width = ov2685_res[res_index].width;
+	fmt->height = ov2685_res[res_index].height;
 	fmt->code = V4L2_MBUS_FMT_UYVY8_1X16;
 
 	mutex_unlock(&dev->input_lock);
diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov2685.h b/drivers/external_drivers/camera/drivers/media/i2c/ov2685.h
index ae0afb8..ad33756 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov2685.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov2685.h
@@ -252,12 +252,62 @@ struct ov2685_control {
 
 /* Supported resolutions */
 enum {
-	OV2685_RES_720P,
+	OV2685_RES_XCIF,
+	OV2685_RES_SVGA,
+	OV2685_RES_1M3,
 	OV2685_RES_2M,
 };
 
 static struct ov2685_res_struct ov2685_res[] = {
 	{
+	.desc	= "XCIF",
+	.res	= OV2685_RES_XCIF,
+	.width	= 732,
+	.height	= 600,
+	.pixels_per_line = 1700,
+	.lines_per_frame = 1294,
+	.fps	= 30,
+	.pix_clk = 66,
+	.used	= 0,
+	.regs	= NULL,
+	.skip_frames = 0,
+	.lanes = 2,
+	.bin_mode = 1,
+	.bin_factor_x = 1,
+	},
+	{
+	.desc	= "SVGA",
+	.res	= OV2685_RES_SVGA,
+	.width	= 800,
+	.height	= 600,
+	.pixels_per_line = 1700,
+	.lines_per_frame = 1294,
+	.fps	= 30,
+	.pix_clk = 66,
+	.used	= 0,
+	.regs	= NULL,
+	.skip_frames = 0,
+	.lanes = 2,
+	.bin_mode = 1,
+	.bin_factor_x = 1,
+	},
+	{
+	.desc	= "1M3",
+	.res	= OV2685_RES_1M3,
+	.width	= 1600,
+	.height	= 900,
+	.pixels_per_line = 1700,
+	.lines_per_frame = 1294,
+	.fps	= 30,
+	.pix_clk = 66,
+	.used	= 0,
+	.regs	= NULL,
+	.skip_frames = 0,
+	.lanes = 2,
+	.bin_mode = 0,
+	.bin_factor_x = 0,
+	},
+	{
 	.desc	= "2M",
 	.res	= OV2685_RES_2M,
 	.width	= 1600,
@@ -311,7 +361,6 @@ static struct ov2685_reg const ov2685_AWB_sunny[] = {
 	{OV2685_TOK_TERM, 0, 0}
 };
 
-
 static struct ov2685_reg const ov2685_AWB_incandescent[] = {
 	{OV2685_8BIT, 0x3208 , 0x00},
 	{OV2685_8BIT, 0x5180 , 0xf6},
@@ -419,6 +468,7 @@ static struct ov2685_reg const ov2685_green_effect[] = {
 	{OV2685_TOK_TERM, 0, 0}
 };
 
+/* resolution: 1600 x 1200, ratios: 4:3, 2 lanes, 30fps */
 static struct ov2685_reg const ov2685_2M_init[] = {
 	/*2lanes, 30fps*/
 	{OV2685_8BIT, 0x0103 , 0x01},
@@ -484,6 +534,7 @@ static struct ov2685_reg const ov2685_2M_init[] = {
 	{OV2685_8BIT, 0x3819 , 0x04},
 	{OV2685_8BIT, 0x3820 , 0xc0},
 	{OV2685_8BIT, 0x3821 , 0x00},
+	/* AIQ settings */
 	{OV2685_8BIT, 0x3a06 , 0x01},
 	{OV2685_8BIT, 0x3a07 , 0x84},
 	{OV2685_8BIT, 0x3a08 , 0x01},
@@ -668,82 +719,86 @@ static struct ov2685_reg const ov2685_2M_init[] = {
 	{OV2685_TOK_TERM, 0, 0}
 };
 
-static struct ov2685_reg const ov2685_720p_init[] = {
-	/*1lane 30fps*/
+/* resolution: 1200 x 900, ratio: 16:9, 2 lanes, 30fps */
+static struct ov2685_reg const ov2685_1M3_init[] = {
+	/*2lanes, 30fps*/
 	{OV2685_8BIT, 0x0103 , 0x01},
 	{OV2685_8BIT, 0x3002 , 0x00},
 	{OV2685_8BIT, 0x3016 , 0x1c},
-	{OV2685_8BIT, 0x3018 , 0x44},
+	{OV2685_8BIT, 0x3018 , 0x84},
 	{OV2685_8BIT, 0x301d , 0xf0},
 	{OV2685_8BIT, 0x3020 , 0x00},
-	{OV2685_8BIT, 0x3082 , 0x37},
+	{OV2685_8BIT, 0x3082 , 0x37},/*mclk = 19.2Mhz*/
 	{OV2685_8BIT, 0x3083 , 0x03},
-	{OV2685_8BIT, 0x3084 , 0x0f},
+	{OV2685_8BIT, 0x3084 , 0x07},
 	{OV2685_8BIT, 0x3085 , 0x03},
 	{OV2685_8BIT, 0x3086 , 0x00},
 	{OV2685_8BIT, 0x3087 , 0x00},
-	{OV2685_8BIT, 0x3501 , 0x2d},
-	{OV2685_8BIT, 0x3502 , 0x80},
+	{OV2685_8BIT, 0x3501 , 0x4e},
+	{OV2685_8BIT, 0x3502 , 0xe0},
 	{OV2685_8BIT, 0x3503 , 0x03},
 	{OV2685_8BIT, 0x350b , 0x36},
 	{OV2685_8BIT, 0x3600 , 0xb4},
 	{OV2685_8BIT, 0x3603 , 0x35},
 	{OV2685_8BIT, 0x3604 , 0x24},
 	{OV2685_8BIT, 0x3605 , 0x00},
-	{OV2685_8BIT, 0x3620 , 0x26},
-	{OV2685_8BIT, 0x3621 , 0x37},
-	{OV2685_8BIT, 0x3622 , 0x04},
+	{OV2685_8BIT, 0x3620 , 0x24},
+	{OV2685_8BIT, 0x3621 , 0x34},
+	{OV2685_8BIT, 0x3622 , 0x03},
 	{OV2685_8BIT, 0x3628 , 0x10},
 	{OV2685_8BIT, 0x3705 , 0x3c},
 	{OV2685_8BIT, 0x370a , 0x21},
 	{OV2685_8BIT, 0x370c , 0x50},
 	{OV2685_8BIT, 0x370d , 0xc0},
 	{OV2685_8BIT, 0x3717 , 0x58},
-	{OV2685_8BIT, 0x3718 , 0x88},
+	{OV2685_8BIT, 0x3718 , 0x80},
 	{OV2685_8BIT, 0x3720 , 0x00},
-	{OV2685_8BIT, 0x3721 , 0x00},
-	{OV2685_8BIT, 0x3722 , 0x00},
-	{OV2685_8BIT, 0x3723 , 0x00},
-	{OV2685_8BIT, 0x3738 , 0x00},
+	{OV2685_8BIT, 0x3721 , 0x09},
+	{OV2685_8BIT, 0x3722 , 0x06},
+	{OV2685_8BIT, 0x3723 , 0x59},
+	{OV2685_8BIT, 0x3738 , 0x99},
 	{OV2685_8BIT, 0x3781 , 0x80},
+	{OV2685_8BIT, 0x3784 , 0x0c},
 	{OV2685_8BIT, 0x3789 , 0x60},
-	{OV2685_8BIT, 0x3800 , 0x00},
-	{OV2685_8BIT, 0x3801 , 0xa0},
-	{OV2685_8BIT, 0x3802 , 0x00},
-	{OV2685_8BIT, 0x3803 , 0xf2},
-	{OV2685_8BIT, 0x3804 , 0x05},
-	{OV2685_8BIT, 0x3805 , 0xaf},
-	{OV2685_8BIT, 0x3806 , 0x03},
-	{OV2685_8BIT, 0x3807 , 0xcd},
-	{OV2685_8BIT, 0x3808 , 0x05},
-	{OV2685_8BIT, 0x3809 , 0x00},
-	{OV2685_8BIT, 0x380a , 0x02},
-	{OV2685_8BIT, 0x380b , 0xd0},
-	{OV2685_8BIT, 0x380c , 0x05},
-	{OV2685_8BIT, 0x380d , 0xa6},
-	{OV2685_8BIT, 0x380e , 0x02},
-	{OV2685_8BIT, 0x380f , 0xf8},
-	{OV2685_8BIT, 0x3810 , 0x00},
+	/* output size settings */
+	{OV2685_8BIT, 0x3800 , 0x00},/* xstart:0 */
+	{OV2685_8BIT, 0x3801 , 0x00},
+	{OV2685_8BIT, 0x3802 , 0x00},/* ystart:150 */
+	{OV2685_8BIT, 0x3803 , 0x96},
+	{OV2685_8BIT, 0x3804 , 0x06},/* xend:1516 */
+	{OV2685_8BIT, 0x3805 , 0x4f},
+	{OV2685_8BIT, 0x3806 , 0x04},/* yend 1065 */
+	{OV2685_8BIT, 0x3807 , 0x29},
+	{OV2685_8BIT, 0x3808 , 0x06},/* xout: 1600 */
+	{OV2685_8BIT, 0x3809 , 0x40},
+	{OV2685_8BIT, 0x380a , 0x03},/* yout: 900 */
+	{OV2685_8BIT, 0x380b , 0x84},
+	{OV2685_8BIT, 0x380c , 0x06},/* hts, vts */
+	{OV2685_8BIT, 0x380d , 0xa4},
+	{OV2685_8BIT, 0x380e , 0x05},
+	{OV2685_8BIT, 0x380f , 0x0e},
+	{OV2685_8BIT, 0x3810 , 0x00},/* xwin:8 */
 	{OV2685_8BIT, 0x3811 , 0x08},
-	{OV2685_8BIT, 0x3812 , 0x00},
-	{OV2685_8BIT, 0x3813 , 0x06},
-	{OV2685_8BIT, 0x3814 , 0x11},
+	{OV2685_8BIT, 0x3812 , 0x00},/* ywin:8 */
+	{OV2685_8BIT, 0x3813 , 0x08},
+	{OV2685_8BIT, 0x3814 , 0x11},/* odd/even inc */
 	{OV2685_8BIT, 0x3815 , 0x11},
-	{OV2685_8BIT, 0x3819 , 0x04},
+	{OV2685_8BIT, 0x3819 , 0x04},/* bin off */
 	{OV2685_8BIT, 0x3820 , 0xc0},
 	{OV2685_8BIT, 0x3821 , 0x00},
-	{OV2685_8BIT, 0x3a06 , 0x00},
-	{OV2685_8BIT, 0x3a07 , 0xe4},
-	{OV2685_8BIT, 0x3a08 , 0x00},
-	{OV2685_8BIT, 0x3a09 , 0xbe},
-	{OV2685_8BIT, 0x3a0a , 0x15},
+	/* AIQ settings */
+	{OV2685_8BIT, 0x3a06 , 0x01},
+	{OV2685_8BIT, 0x3a07 , 0x84},
+	{OV2685_8BIT, 0x3a08 , 0x01},
+	{OV2685_8BIT, 0x3a09 , 0x43},
+	{OV2685_8BIT, 0x3a0a , 0x24},
 	{OV2685_8BIT, 0x3a0b , 0x60},
-	{OV2685_8BIT, 0x3a0c , 0x17},
-	{OV2685_8BIT, 0x3a0d , 0xc0},
-	{OV2685_8BIT, 0x3a0e , 0x02},
-	{OV2685_8BIT, 0x3a0f , 0xac},
-	{OV2685_8BIT, 0x3a10 , 0x02},
-	{OV2685_8BIT, 0x3a11 , 0xf8},
+	{OV2685_8BIT, 0x3a0c , 0x28},
+	{OV2685_8BIT, 0x3a0d , 0x60},
+	{OV2685_8BIT, 0x3a0e , 0x04},
+	{OV2685_8BIT, 0x3a0f , 0x8c},
+	{OV2685_8BIT, 0x3a10 , 0x05},
+	{OV2685_8BIT, 0x3a11 , 0x0c},
 	{OV2685_8BIT, 0x4000 , 0x81},
 	{OV2685_8BIT, 0x4001 , 0x40},
 	{OV2685_8BIT, 0x4008 , 0x02},
@@ -916,86 +971,90 @@ static struct ov2685_reg const ov2685_720p_init[] = {
 	{OV2685_TOK_TERM, 0, 0}
 };
 
-/* camera vga 30fps, yuv, 1lanes */
-static struct ov2685_reg const ov2685_vga_init[] = {
+/* resolution: 1408 x 1152, ratio: 11:9, 2 lanes, 30fps */
+static struct ov2685_reg const ov2685_1408_1152_init[] = {
+	/*2lanes, 30fps*/
 	{OV2685_8BIT, 0x0103 , 0x01},
 	{OV2685_8BIT, 0x3002 , 0x00},
 	{OV2685_8BIT, 0x3016 , 0x1c},
-	{OV2685_8BIT, 0x3018 , 0x44},
+	{OV2685_8BIT, 0x3018 , 0x84},
 	{OV2685_8BIT, 0x301d , 0xf0},
 	{OV2685_8BIT, 0x3020 , 0x00},
-	{OV2685_8BIT, 0x3082 , 0x37},
+	{OV2685_8BIT, 0x3082 , 0x37},/*mclk = 19.2Mhz*/
 	{OV2685_8BIT, 0x3083 , 0x03},
-	{OV2685_8BIT, 0x3084 , 0x0f},
+	{OV2685_8BIT, 0x3084 , 0x07},
 	{OV2685_8BIT, 0x3085 , 0x03},
 	{OV2685_8BIT, 0x3086 , 0x00},
 	{OV2685_8BIT, 0x3087 , 0x00},
-	{OV2685_8BIT, 0x3501 , 0x26},
-	{OV2685_8BIT, 0x3502 , 0x40},
+	{OV2685_8BIT, 0x3501 , 0x4e},
+	{OV2685_8BIT, 0x3502 , 0xe0},
 	{OV2685_8BIT, 0x3503 , 0x03},
 	{OV2685_8BIT, 0x350b , 0x36},
 	{OV2685_8BIT, 0x3600 , 0xb4},
 	{OV2685_8BIT, 0x3603 , 0x35},
 	{OV2685_8BIT, 0x3604 , 0x24},
 	{OV2685_8BIT, 0x3605 , 0x00},
-	{OV2685_8BIT, 0x3620 , 0x26},
-	{OV2685_8BIT, 0x3621 , 0x37},
-	{OV2685_8BIT, 0x3622 , 0x04},
+	{OV2685_8BIT, 0x3620 , 0x24},
+	{OV2685_8BIT, 0x3621 , 0x34},
+	{OV2685_8BIT, 0x3622 , 0x03},
 	{OV2685_8BIT, 0x3628 , 0x10},
 	{OV2685_8BIT, 0x3705 , 0x3c},
-	{OV2685_8BIT, 0x370a , 0x23},
+	{OV2685_8BIT, 0x370a , 0x21},
 	{OV2685_8BIT, 0x370c , 0x50},
 	{OV2685_8BIT, 0x370d , 0xc0},
 	{OV2685_8BIT, 0x3717 , 0x58},
-	{OV2685_8BIT, 0x3718 , 0x88},
+	{OV2685_8BIT, 0x3718 , 0x80},
 	{OV2685_8BIT, 0x3720 , 0x00},
-	{OV2685_8BIT, 0x3721 , 0x00},
-	{OV2685_8BIT, 0x3722 , 0x00},
-	{OV2685_8BIT, 0x3723 , 0x00},
-	{OV2685_8BIT, 0x3738 , 0x00},
+	{OV2685_8BIT, 0x3721 , 0x09},
+	{OV2685_8BIT, 0x3722 , 0x06},
+	{OV2685_8BIT, 0x3723 , 0x59},
+	{OV2685_8BIT, 0x3738 , 0x99},
 	{OV2685_8BIT, 0x3781 , 0x80},
+	{OV2685_8BIT, 0x3784 , 0x0c},
 	{OV2685_8BIT, 0x3789 , 0x60},
-	{OV2685_8BIT, 0x3800 , 0x00},
-	{OV2685_8BIT, 0x3801 , 0xa0},
-	{OV2685_8BIT, 0x3802 , 0x00},
-	{OV2685_8BIT, 0x3803 , 0x78},
-	{OV2685_8BIT, 0x3804 , 0x05},
-	{OV2685_8BIT, 0x3805 , 0xaf},
-	{OV2685_8BIT, 0x3806 , 0x04},
-	{OV2685_8BIT, 0x3807 , 0x47},
-	{OV2685_8BIT, 0x3808 , 0x02},
+	/* output size settings */
+	{OV2685_8BIT, 0x3800 , 0x00},/* xstart:96 */
+	{OV2685_8BIT, 0x3801 , 0x60},
+	{OV2685_8BIT, 0x3802 , 0x00},/* ystart:24 */
+	{OV2685_8BIT, 0x3803 , 0x18},
+	{OV2685_8BIT, 0x3804 , 0x04},/* xend:1119 */
+	{OV2685_8BIT, 0x3805 , 0x5f},
+	{OV2685_8BIT, 0x3806 , 0x04},/* yend:1191 */
+	{OV2685_8BIT, 0x3807 , 0xa7},
+	{OV2685_8BIT, 0x3808 , 0x05},/* xout:1408 */
 	{OV2685_8BIT, 0x3809 , 0x80},
-	{OV2685_8BIT, 0x380a , 0x01},
-	{OV2685_8BIT, 0x380b , 0xe0},
+	{OV2685_8BIT, 0x380a , 0x04},/* yout:1152 */
+	{OV2685_8BIT, 0x380b , 0x80},
 	{OV2685_8BIT, 0x380c , 0x06},
-	{OV2685_8BIT, 0x380d , 0xac},
-	{OV2685_8BIT, 0x380e , 0x02},
-	{OV2685_8BIT, 0x380f , 0x84},
+	{OV2685_8BIT, 0x380d , 0xa4},
+	{OV2685_8BIT, 0x380e , 0x05},
+	{OV2685_8BIT, 0x380f , 0x0e},
 	{OV2685_8BIT, 0x3810 , 0x00},
-	{OV2685_8BIT, 0x3811 , 0x04},
+	{OV2685_8BIT, 0x3811 , 0x08},
 	{OV2685_8BIT, 0x3812 , 0x00},
-	{OV2685_8BIT, 0x3813 , 0x04},
-	{OV2685_8BIT, 0x3814 , 0x31},
-	{OV2685_8BIT, 0x3815 , 0x31},
+	{OV2685_8BIT, 0x3813 , 0x08},
+	{OV2685_8BIT, 0x3814 , 0x11},
+	{OV2685_8BIT, 0x3815 , 0x11},
 	{OV2685_8BIT, 0x3819 , 0x04},
-	{OV2685_8BIT, 0x3820 , 0xc2},
-	{OV2685_8BIT, 0x3821 , 0x01},
-	{OV2685_8BIT, 0x3a06 , 0x00},
-	{OV2685_8BIT, 0x3a07 , 0xc1},
-	{OV2685_8BIT, 0x3a08 , 0x00},
-	{OV2685_8BIT, 0x3a09 , 0xa1},
-	{OV2685_8BIT, 0x3a0a , 0x12},
-	{OV2685_8BIT, 0x3a0b , 0x18},
-	{OV2685_8BIT, 0x3a0c , 0x14},
-	{OV2685_8BIT, 0x3a0d , 0x20},
-	{OV2685_8BIT, 0x3a0e , 0x02},
-	{OV2685_8BIT, 0x3a0f , 0x43},
-	{OV2685_8BIT, 0x3a10 , 0x02},
-	{OV2685_8BIT, 0x3a11 , 0x84},
+	{OV2685_8BIT, 0x3820 , 0xc0},
+	/* AIQ settings */
+	{OV2685_8BIT, 0x3821 , 0x00},
+	{OV2685_8BIT, 0x3a06 , 0x01},
+	{OV2685_8BIT, 0x3a07 , 0x84},
+	{OV2685_8BIT, 0x3a08 , 0x01},
+	{OV2685_8BIT, 0x3a09 , 0x43},
+	{OV2685_8BIT, 0x3a0a , 0x24},
+	{OV2685_8BIT, 0x3a0b , 0x60},
+	{OV2685_8BIT, 0x3a0c , 0x28},
+	{OV2685_8BIT, 0x3a0d , 0x60},
+	{OV2685_8BIT, 0x3a0e , 0x04},
+	{OV2685_8BIT, 0x3a0f , 0x8c},
+	{OV2685_8BIT, 0x3a10 , 0x05},
+	{OV2685_8BIT, 0x3a11 , 0x0c},
 	{OV2685_8BIT, 0x4000 , 0x81},
 	{OV2685_8BIT, 0x4001 , 0x40},
-	{OV2685_8BIT, 0x4008 , 0x00},
-	{OV2685_8BIT, 0x4009 , 0x03},
+	{OV2685_8BIT, 0x4008 , 0x02},
+	{OV2685_8BIT, 0x4009 , 0x09},
 	{OV2685_8BIT, 0x4300 , 0x32},
 	{OV2685_8BIT, 0x430e , 0x00},
 	{OV2685_8BIT, 0x4602 , 0x02},
@@ -1007,9 +1066,6 @@ static struct ov2685_reg const ov2685_vga_init[] = {
 	{OV2685_8BIT, 0x5004 , 0xff},
 	{OV2685_8BIT, 0x5005 , 0x12},
 	{OV2685_8BIT, 0x0100 , 0x01},
-	{OV2685_8BIT, 0x0101 , 0x01},
-	{OV2685_8BIT, 0x1000 , 0x01},
-	{OV2685_8BIT, 0x0129 , 0x10},
 	{OV2685_8BIT, 0x5180 , 0xf4},
 	{OV2685_8BIT, 0x5181 , 0x11},
 	{OV2685_8BIT, 0x5182 , 0x41},
@@ -1167,6 +1223,1020 @@ static struct ov2685_reg const ov2685_vga_init[] = {
 	{OV2685_TOK_TERM, 0, 0}
 };
 
+/*
+ * resolution: 1468 x 1200, ratio: 11:9, 2 lanes, 30fps
+ * height: full size.
+ */
+static struct ov2685_reg const ov2685_1468_1200_init[] = {
+	/*2lanes, 30fps*/
+	{OV2685_8BIT, 0x0103 , 0x01},
+	{OV2685_8BIT, 0x3002 , 0x00},
+	{OV2685_8BIT, 0x3016 , 0x1c},
+	{OV2685_8BIT, 0x3018 , 0x84},
+	{OV2685_8BIT, 0x301d , 0xf0},
+	{OV2685_8BIT, 0x3020 , 0x00},
+	{OV2685_8BIT, 0x3082 , 0x37},/*mclk = 19.2Mhz*/
+	{OV2685_8BIT, 0x3083 , 0x03},
+	{OV2685_8BIT, 0x3084 , 0x07},
+	{OV2685_8BIT, 0x3085 , 0x03},
+	{OV2685_8BIT, 0x3086 , 0x00},
+	{OV2685_8BIT, 0x3087 , 0x00},
+	{OV2685_8BIT, 0x3501 , 0x4e},
+	{OV2685_8BIT, 0x3502 , 0xe0},
+	{OV2685_8BIT, 0x3503 , 0x03},
+	{OV2685_8BIT, 0x350b , 0x36},
+	{OV2685_8BIT, 0x3600 , 0xb4},
+	{OV2685_8BIT, 0x3603 , 0x35},
+	{OV2685_8BIT, 0x3604 , 0x24},
+	{OV2685_8BIT, 0x3605 , 0x00},
+	{OV2685_8BIT, 0x3620 , 0x24},
+	{OV2685_8BIT, 0x3621 , 0x34},
+	{OV2685_8BIT, 0x3622 , 0x03},
+	{OV2685_8BIT, 0x3628 , 0x10},
+	{OV2685_8BIT, 0x3705 , 0x3c},
+	{OV2685_8BIT, 0x370a , 0x21},
+	{OV2685_8BIT, 0x370c , 0x50},
+	{OV2685_8BIT, 0x370d , 0xc0},
+	{OV2685_8BIT, 0x3717 , 0x58},
+	{OV2685_8BIT, 0x3718 , 0x80},
+	{OV2685_8BIT, 0x3720 , 0x00},
+	{OV2685_8BIT, 0x3721 , 0x09},
+	{OV2685_8BIT, 0x3722 , 0x06},
+	{OV2685_8BIT, 0x3723 , 0x59},
+	{OV2685_8BIT, 0x3738 , 0x99},
+	{OV2685_8BIT, 0x3781 , 0x80},
+	{OV2685_8BIT, 0x3784 , 0x0c},
+	{OV2685_8BIT, 0x3789 , 0x60},
+	/* output size settings */
+	{OV2685_8BIT, 0x3800 , 0x00}, /* xstart:66 */
+	{OV2685_8BIT, 0x3801 , 0x42},
+	{OV2685_8BIT, 0x3802 , 0x00}, /* ystart:0 */
+	{OV2685_8BIT, 0x3803 , 0x00},
+	{OV2685_8BIT, 0x3804 , 0x06}, /* xend:1549 */
+	{OV2685_8BIT, 0x3805 , 0x0d},
+	{OV2685_8BIT, 0x3806 , 0x04}, /* yend:1215 */
+	{OV2685_8BIT, 0x3807 , 0xbf},
+	{OV2685_8BIT, 0x3808 , 0x05}, /* xout:1468 */
+	{OV2685_8BIT, 0x3809 , 0xbc},
+	{OV2685_8BIT, 0x380a , 0x04}, /* yout:1200 */
+	{OV2685_8BIT, 0x380b , 0xb0},
+	{OV2685_8BIT, 0x380c , 0x06},
+	{OV2685_8BIT, 0x380d , 0xa4},
+	{OV2685_8BIT, 0x380e , 0x05},
+	{OV2685_8BIT, 0x380f , 0x0e},
+	{OV2685_8BIT, 0x3810 , 0x00},
+	{OV2685_8BIT, 0x3811 , 0x08},
+	{OV2685_8BIT, 0x3812 , 0x00},
+	{OV2685_8BIT, 0x3813 , 0x08},
+	{OV2685_8BIT, 0x3814 , 0x11},
+	{OV2685_8BIT, 0x3815 , 0x11},
+	{OV2685_8BIT, 0x3819 , 0x04},
+	{OV2685_8BIT, 0x3820 , 0xc0},
+	{OV2685_8BIT, 0x3821 , 0x00},
+	/* AIQ settings */
+	{OV2685_8BIT, 0x3a06 , 0x01},
+	{OV2685_8BIT, 0x3a07 , 0x84},
+	{OV2685_8BIT, 0x3a08 , 0x01},
+	{OV2685_8BIT, 0x3a09 , 0x43},
+	{OV2685_8BIT, 0x3a0a , 0x24},
+	{OV2685_8BIT, 0x3a0b , 0x60},
+	{OV2685_8BIT, 0x3a0c , 0x28},
+	{OV2685_8BIT, 0x3a0d , 0x60},
+	{OV2685_8BIT, 0x3a0e , 0x04},
+	{OV2685_8BIT, 0x3a0f , 0x8c},
+	{OV2685_8BIT, 0x3a10 , 0x05},
+	{OV2685_8BIT, 0x3a11 , 0x0c},
+	{OV2685_8BIT, 0x4000 , 0x81},
+	{OV2685_8BIT, 0x4001 , 0x40},
+	{OV2685_8BIT, 0x4008 , 0x02},
+	{OV2685_8BIT, 0x4009 , 0x09},
+	{OV2685_8BIT, 0x4300 , 0x32},
+	{OV2685_8BIT, 0x430e , 0x00},
+	{OV2685_8BIT, 0x4602 , 0x02},
+	{OV2685_8BIT, 0x4837 , 0x1e},
+	{OV2685_8BIT, 0x5000 , 0xff},
+	{OV2685_8BIT, 0x5001 , 0x05},
+	{OV2685_8BIT, 0x5002 , 0x32},
+	{OV2685_8BIT, 0x5003 , 0x04},
+	{OV2685_8BIT, 0x5004 , 0xff},
+	{OV2685_8BIT, 0x5005 , 0x12},
+	{OV2685_8BIT, 0x0100 , 0x01},
+	{OV2685_8BIT, 0x5180 , 0xf4},
+	{OV2685_8BIT, 0x5181 , 0x11},
+	{OV2685_8BIT, 0x5182 , 0x41},
+	{OV2685_8BIT, 0x5183 , 0x42},
+	{OV2685_8BIT, 0x5184 , 0x78},
+	{OV2685_8BIT, 0x5185 , 0x58},
+	{OV2685_8BIT, 0x5186 , 0xb5},
+	{OV2685_8BIT, 0x5187 , 0xb2},
+	{OV2685_8BIT, 0x5188 , 0x08},
+	{OV2685_8BIT, 0x5189 , 0x0e},
+	{OV2685_8BIT, 0x518a , 0x0c},
+	{OV2685_8BIT, 0x518b , 0x4c},
+	{OV2685_8BIT, 0x518c , 0x38},
+	{OV2685_8BIT, 0x518d , 0xf8},
+	{OV2685_8BIT, 0x518e , 0x04},
+	{OV2685_8BIT, 0x518f , 0x7f},
+	{OV2685_8BIT, 0x5190 , 0x40},
+	{OV2685_8BIT, 0x5191 , 0x5f},
+	{OV2685_8BIT, 0x5192 , 0x40},
+	{OV2685_8BIT, 0x5193 , 0xff},
+	{OV2685_8BIT, 0x5194 , 0x40},
+	{OV2685_8BIT, 0x5195 , 0x07},
+	{OV2685_8BIT, 0x5196 , 0x04},
+	{OV2685_8BIT, 0x5197 , 0x04},
+	{OV2685_8BIT, 0x5198 , 0x00},
+	{OV2685_8BIT, 0x5199 , 0x05},
+	{OV2685_8BIT, 0x519a , 0xd2},
+	{OV2685_8BIT, 0x519b , 0x10},
+	{OV2685_8BIT, 0x5200 , 0x09},
+	{OV2685_8BIT, 0x5201 , 0x00},
+	{OV2685_8BIT, 0x5202 , 0x06},
+	{OV2685_8BIT, 0x5203 , 0x20},
+	{OV2685_8BIT, 0x5204 , 0x41},
+	{OV2685_8BIT, 0x5205 , 0x16},
+	{OV2685_8BIT, 0x5206 , 0x00},
+	{OV2685_8BIT, 0x5207 , 0x05},
+	{OV2685_8BIT, 0x520b , 0x30},
+	{OV2685_8BIT, 0x520c , 0x75},
+	{OV2685_8BIT, 0x520d , 0x00},
+	{OV2685_8BIT, 0x520e , 0x30},
+	{OV2685_8BIT, 0x520f , 0x75},
+	{OV2685_8BIT, 0x5210 , 0x00},
+	{OV2685_8BIT, 0x5280 , 0x14},
+	{OV2685_8BIT, 0x5281 , 0x02},
+	{OV2685_8BIT, 0x5282 , 0x02},
+	{OV2685_8BIT, 0x5283 , 0x04},
+	{OV2685_8BIT, 0x5284 , 0x06},
+	{OV2685_8BIT, 0x5285 , 0x08},
+	{OV2685_8BIT, 0x5286 , 0x0c},
+	{OV2685_8BIT, 0x5287 , 0x10},
+	{OV2685_8BIT, 0x5300 , 0xc5},
+	{OV2685_8BIT, 0x5301 , 0xa0},
+	{OV2685_8BIT, 0x5302 , 0x06},
+	{OV2685_8BIT, 0x5303 , 0x0a},
+	{OV2685_8BIT, 0x5304 , 0x30},
+	{OV2685_8BIT, 0x5305 , 0x60},
+	{OV2685_8BIT, 0x5306 , 0x90},
+	{OV2685_8BIT, 0x5307 , 0xc0},
+	{OV2685_8BIT, 0x5308 , 0x82},
+	{OV2685_8BIT, 0x5309 , 0x00},
+	{OV2685_8BIT, 0x530a , 0x26},
+	{OV2685_8BIT, 0x530b , 0x02},
+	{OV2685_8BIT, 0x530c , 0x02},
+	{OV2685_8BIT, 0x530d , 0x00},
+	{OV2685_8BIT, 0x530e , 0x0c},
+	{OV2685_8BIT, 0x530f , 0x14},
+	{OV2685_8BIT, 0x5310 , 0x1a},
+	{OV2685_8BIT, 0x5311 , 0x20},
+	{OV2685_8BIT, 0x5312 , 0x80},
+	{OV2685_8BIT, 0x5313 , 0x4b},
+	{OV2685_8BIT, 0x5380 , 0x01},
+	{OV2685_8BIT, 0x5381 , 0x52},
+	{OV2685_8BIT, 0x5382 , 0x00},
+	{OV2685_8BIT, 0x5383 , 0x4a},
+	{OV2685_8BIT, 0x5384 , 0x00},
+	{OV2685_8BIT, 0x5385 , 0xb6},
+	{OV2685_8BIT, 0x5386 , 0x00},
+	{OV2685_8BIT, 0x5387 , 0x8d},
+	{OV2685_8BIT, 0x5388 , 0x00},
+	{OV2685_8BIT, 0x5389 , 0x3a},
+	{OV2685_8BIT, 0x538a , 0x00},
+	{OV2685_8BIT, 0x538b , 0xa6},
+	{OV2685_8BIT, 0x538c , 0x00},
+	{OV2685_8BIT, 0x5400 , 0x0d},
+	{OV2685_8BIT, 0x5401 , 0x18},
+	{OV2685_8BIT, 0x5402 , 0x31},
+	{OV2685_8BIT, 0x5403 , 0x5a},
+	{OV2685_8BIT, 0x5404 , 0x65},
+	{OV2685_8BIT, 0x5405 , 0x6f},
+	{OV2685_8BIT, 0x5406 , 0x77},
+	{OV2685_8BIT, 0x5407 , 0x80},
+	{OV2685_8BIT, 0x5408 , 0x87},
+	{OV2685_8BIT, 0x5409 , 0x8f},
+	{OV2685_8BIT, 0x540a , 0xa2},
+	{OV2685_8BIT, 0x540b , 0xb2},
+	{OV2685_8BIT, 0x540c , 0xcc},
+	{OV2685_8BIT, 0x540d , 0xe4},
+	{OV2685_8BIT, 0x540e , 0xf0},
+	{OV2685_8BIT, 0x540f , 0xa0},
+	{OV2685_8BIT, 0x5410 , 0x6e},
+	{OV2685_8BIT, 0x5411 , 0x06},
+	{OV2685_8BIT, 0x5480 , 0x19},
+	{OV2685_8BIT, 0x5481 , 0x00},
+	{OV2685_8BIT, 0x5482 , 0x09},
+	{OV2685_8BIT, 0x5483 , 0x12},
+	{OV2685_8BIT, 0x5484 , 0x04},
+	{OV2685_8BIT, 0x5485 , 0x06},
+	{OV2685_8BIT, 0x5486 , 0x08},
+	{OV2685_8BIT, 0x5487 , 0x0c},
+	{OV2685_8BIT, 0x5488 , 0x10},
+	{OV2685_8BIT, 0x5489 , 0x18},
+	{OV2685_8BIT, 0x5500 , 0x02},
+	{OV2685_8BIT, 0x5501 , 0x03},
+	{OV2685_8BIT, 0x5502 , 0x04},
+	{OV2685_8BIT, 0x5503 , 0x05},
+	{OV2685_8BIT, 0x5504 , 0x06},
+	{OV2685_8BIT, 0x5505 , 0x08},
+	{OV2685_8BIT, 0x5506 , 0x00},
+	{OV2685_8BIT, 0x5600 , 0x02},
+	{OV2685_8BIT, 0x5603 , 0x40},
+	{OV2685_8BIT, 0x5604 , 0x28},
+	{OV2685_8BIT, 0x5609 , 0x20},
+	{OV2685_8BIT, 0x560a , 0x60},
+	{OV2685_8BIT, 0x5800 , 0x03},
+	{OV2685_8BIT, 0x5801 , 0x24},
+	{OV2685_8BIT, 0x5802 , 0x02},
+	{OV2685_8BIT, 0x5803 , 0x40},
+	{OV2685_8BIT, 0x5804 , 0x34},
+	{OV2685_8BIT, 0x5805 , 0x05},
+	{OV2685_8BIT, 0x5806 , 0x12},
+	{OV2685_8BIT, 0x5807 , 0x05},
+	{OV2685_8BIT, 0x5808 , 0x03},
+	{OV2685_8BIT, 0x5809 , 0x3c},
+	{OV2685_8BIT, 0x580a , 0x02},
+	{OV2685_8BIT, 0x580b , 0x40},
+	{OV2685_8BIT, 0x580c , 0x26},
+	{OV2685_8BIT, 0x580d , 0x05},
+	{OV2685_8BIT, 0x580e , 0x52},
+	{OV2685_8BIT, 0x580f , 0x06},
+	{OV2685_8BIT, 0x5810 , 0x03},
+	{OV2685_8BIT, 0x5811 , 0x28},
+	{OV2685_8BIT, 0x5812 , 0x02},
+	{OV2685_8BIT, 0x5813 , 0x40},
+	{OV2685_8BIT, 0x5814 , 0x24},
+	{OV2685_8BIT, 0x5815 , 0x05},
+	{OV2685_8BIT, 0x5816 , 0x42},
+	{OV2685_8BIT, 0x5817 , 0x06},
+	{OV2685_8BIT, 0x5818 , 0x0d},
+	{OV2685_8BIT, 0x5819 , 0x40},
+	{OV2685_8BIT, 0x581a , 0x04},
+	{OV2685_8BIT, 0x581b , 0x0c},
+	{OV2685_8BIT, 0x3a03 , 0x4c},
+	{OV2685_8BIT, 0x3a04 , 0x40},
+	{OV2685_8BIT, 0x3503 , 0x00},
+	{OV2685_TOK_TERM, 0, 0}
+};
+
+/*
+ * resolution: 732 x 600, ratio: 11:9, 2 lanes, 30fps
+ * height: full size, 2x2 downscaling.
+ */
+static struct ov2685_reg const ov2685_732_600_init[] = {
+	/*2lanes, 30fps*/
+	{OV2685_8BIT, 0x0103 , 0x01},
+	{OV2685_8BIT, 0x3002 , 0x00},
+	{OV2685_8BIT, 0x3016 , 0x1c},
+	{OV2685_8BIT, 0x3018 , 0x84},
+	{OV2685_8BIT, 0x301d , 0xf0},
+	{OV2685_8BIT, 0x3020 , 0x00},
+	{OV2685_8BIT, 0x3082 , 0x37},/*mclk = 19.2Mhz*/
+	{OV2685_8BIT, 0x3083 , 0x03},
+	{OV2685_8BIT, 0x3084 , 0x07},
+	{OV2685_8BIT, 0x3085 , 0x03},
+	{OV2685_8BIT, 0x3086 , 0x00},
+	{OV2685_8BIT, 0x3087 , 0x00},
+	{OV2685_8BIT, 0x3501 , 0x26},
+	{OV2685_8BIT, 0x3502 , 0x40},
+	{OV2685_8BIT, 0x3503 , 0x03},
+	{OV2685_8BIT, 0x350b , 0x36},
+	{OV2685_8BIT, 0x3600 , 0xb4},
+	{OV2685_8BIT, 0x3603 , 0x35},
+	{OV2685_8BIT, 0x3604 , 0x24},
+	{OV2685_8BIT, 0x3605 , 0x00},
+	{OV2685_8BIT, 0x3620 , 0x24},
+	{OV2685_8BIT, 0x3621 , 0x34},
+	{OV2685_8BIT, 0x3622 , 0x03},
+	{OV2685_8BIT, 0x3628 , 0x10},
+	{OV2685_8BIT, 0x3705 , 0x3c},
+	{OV2685_8BIT, 0x370a , 0x21},
+	{OV2685_8BIT, 0x370c , 0x50},
+	{OV2685_8BIT, 0x370d , 0xc0},
+	{OV2685_8BIT, 0x3717 , 0x58},
+	{OV2685_8BIT, 0x3718 , 0x80},
+	{OV2685_8BIT, 0x3720 , 0x00},
+	{OV2685_8BIT, 0x3721 , 0x09},
+	{OV2685_8BIT, 0x3722 , 0x0b},
+	{OV2685_8BIT, 0x3723 , 0x48},
+	{OV2685_8BIT, 0x3738 , 0x99},
+	{OV2685_8BIT, 0x3781 , 0x80},
+	{OV2685_8BIT, 0x3784 , 0x0c},
+	{OV2685_8BIT, 0x3789 , 0x60},
+	/* output size settigs */
+	{OV2685_8BIT, 0x3800 , 0x00},/* xstart:68 */
+	{OV2685_8BIT, 0x3801 , 0x44},
+	{OV2685_8BIT, 0x3802 , 0x00},/* ystart:0 */
+	{OV2685_8BIT, 0x3803 , 0x00},
+	{OV2685_8BIT, 0x3804 , 0x06},/* xend:1547 */
+	{OV2685_8BIT, 0x3805 , 0x0b},
+	{OV2685_8BIT, 0x3806 , 0x04},/* yend:1215 */
+	{OV2685_8BIT, 0x3807 , 0xbf},
+	{OV2685_8BIT, 0x3808 , 0x02},/* xout:732 */
+	{OV2685_8BIT, 0x3809 , 0xdc},
+	{OV2685_8BIT, 0x380a , 0x02},/* yout:600 */
+	{OV2685_8BIT, 0x380b , 0x58},
+	{OV2685_8BIT, 0x380c , 0x06},
+	{OV2685_8BIT, 0x380d , 0xa4},
+	{OV2685_8BIT, 0x380e , 0x05},
+	{OV2685_8BIT, 0x380f , 0x0e},
+	{OV2685_8BIT, 0x3810 , 0x00},
+	{OV2685_8BIT, 0x3811 , 0x04},
+	{OV2685_8BIT, 0x3812 , 0x00},
+	{OV2685_8BIT, 0x3813 , 0x04},
+	{OV2685_8BIT, 0x3814 , 0x31},
+	{OV2685_8BIT, 0x3815 , 0x31},
+	{OV2685_8BIT, 0x3819 , 0x04},
+	{OV2685_8BIT, 0x3820 , 0xc2},
+	{OV2685_8BIT, 0x3821 , 0x01},
+	/* AIQ settings */
+	{OV2685_8BIT, 0x3a06 , 0x01},
+	{OV2685_8BIT, 0x3a07 , 0x84},
+	{OV2685_8BIT, 0x3a08 , 0x01},
+	{OV2685_8BIT, 0x3a09 , 0x43},
+	{OV2685_8BIT, 0x3a0a , 0x24},
+	{OV2685_8BIT, 0x3a0b , 0x60},
+	{OV2685_8BIT, 0x3a0c , 0x28},
+	{OV2685_8BIT, 0x3a0d , 0x60},
+	{OV2685_8BIT, 0x3a0e , 0x04},
+	{OV2685_8BIT, 0x3a0f , 0x8c},
+	{OV2685_8BIT, 0x3a10 , 0x05},
+	{OV2685_8BIT, 0x3a11 , 0x0c},
+	{OV2685_8BIT, 0x3a13 , 0x80},
+	{OV2685_8BIT, 0x4000 , 0x81},
+	{OV2685_8BIT, 0x4001 , 0x40},
+	{OV2685_8BIT, 0x4008 , 0x00},
+	{OV2685_8BIT, 0x4009 , 0x03},
+	{OV2685_8BIT, 0x4300 , 0x32},
+	{OV2685_8BIT, 0x430e , 0x00},
+	{OV2685_8BIT, 0x4602 , 0x02},
+	{OV2685_8BIT, 0x4837 , 0x1e},
+	{OV2685_8BIT, 0x5000 , 0xff},
+	{OV2685_8BIT, 0x5001 , 0x05},
+	{OV2685_8BIT, 0x5002 , 0x32},
+	{OV2685_8BIT, 0x5003 , 0x0c},
+	{OV2685_8BIT, 0x5004 , 0xff},
+	{OV2685_8BIT, 0x5005 , 0x12},
+	{OV2685_8BIT, 0x0100 , 0x01},
+	{OV2685_8BIT, 0x5180 , 0xf4},
+	{OV2685_8BIT, 0x5181 , 0x11},
+	{OV2685_8BIT, 0x5182 , 0x41},
+	{OV2685_8BIT, 0x5183 , 0x42},
+	{OV2685_8BIT, 0x5184 , 0x78},
+	{OV2685_8BIT, 0x5185 , 0x58},
+	{OV2685_8BIT, 0x5186 , 0xb5},
+	{OV2685_8BIT, 0x5187 , 0xb2},
+	{OV2685_8BIT, 0x5188 , 0x08},
+	{OV2685_8BIT, 0x5189 , 0x0e},
+	{OV2685_8BIT, 0x518a , 0x0c},
+	{OV2685_8BIT, 0x518b , 0x4c},
+	{OV2685_8BIT, 0x518c , 0x38},
+	{OV2685_8BIT, 0x518d , 0xf8},
+	{OV2685_8BIT, 0x518e , 0x04},
+	{OV2685_8BIT, 0x518f , 0x7f},
+	{OV2685_8BIT, 0x5190 , 0x40},
+	{OV2685_8BIT, 0x5191 , 0x5f},
+	{OV2685_8BIT, 0x5192 , 0x40},
+	{OV2685_8BIT, 0x5193 , 0xff},
+	{OV2685_8BIT, 0x5194 , 0x40},
+	{OV2685_8BIT, 0x5195 , 0x07},
+	{OV2685_8BIT, 0x5196 , 0x04},
+	{OV2685_8BIT, 0x5197 , 0x04},
+	{OV2685_8BIT, 0x5198 , 0x00},
+	{OV2685_8BIT, 0x5199 , 0x05},
+	{OV2685_8BIT, 0x519a , 0xd2},
+	{OV2685_8BIT, 0x519b , 0x10},
+	{OV2685_8BIT, 0x5200 , 0x09},
+	{OV2685_8BIT, 0x5201 , 0x00},
+	{OV2685_8BIT, 0x5202 , 0x06},
+	{OV2685_8BIT, 0x5203 , 0x20},
+	{OV2685_8BIT, 0x5204 , 0x41},
+	{OV2685_8BIT, 0x5205 , 0x16},
+	{OV2685_8BIT, 0x5206 , 0x00},
+	{OV2685_8BIT, 0x5207 , 0x05},
+	{OV2685_8BIT, 0x520b , 0x30},
+	{OV2685_8BIT, 0x520c , 0x75},
+	{OV2685_8BIT, 0x520d , 0x00},
+	{OV2685_8BIT, 0x520e , 0x30},
+	{OV2685_8BIT, 0x520f , 0x75},
+	{OV2685_8BIT, 0x5210 , 0x00},
+	{OV2685_8BIT, 0x5280 , 0x14},
+	{OV2685_8BIT, 0x5281 , 0x02},
+	{OV2685_8BIT, 0x5282 , 0x02},
+	{OV2685_8BIT, 0x5283 , 0x04},
+	{OV2685_8BIT, 0x5284 , 0x06},
+	{OV2685_8BIT, 0x5285 , 0x08},
+	{OV2685_8BIT, 0x5286 , 0x0c},
+	{OV2685_8BIT, 0x5287 , 0x10},
+	{OV2685_8BIT, 0x5300 , 0xc5},
+	{OV2685_8BIT, 0x5301 , 0xa0},
+	{OV2685_8BIT, 0x5302 , 0x06},
+	{OV2685_8BIT, 0x5303 , 0x0a},
+	{OV2685_8BIT, 0x5304 , 0x30},
+	{OV2685_8BIT, 0x5305 , 0x60},
+	{OV2685_8BIT, 0x5306 , 0x90},
+	{OV2685_8BIT, 0x5307 , 0xc0},
+	{OV2685_8BIT, 0x5308 , 0x82},
+	{OV2685_8BIT, 0x5309 , 0x00},
+	{OV2685_8BIT, 0x530a , 0x26},
+	{OV2685_8BIT, 0x530b , 0x02},
+	{OV2685_8BIT, 0x530c , 0x02},
+	{OV2685_8BIT, 0x530d , 0x00},
+	{OV2685_8BIT, 0x530e , 0x0c},
+	{OV2685_8BIT, 0x530f , 0x14},
+	{OV2685_8BIT, 0x5310 , 0x1a},
+	{OV2685_8BIT, 0x5311 , 0x20},
+	{OV2685_8BIT, 0x5312 , 0x80},
+	{OV2685_8BIT, 0x5313 , 0x4b},
+	{OV2685_8BIT, 0x5380 , 0x01},
+	{OV2685_8BIT, 0x5381 , 0x52},
+	{OV2685_8BIT, 0x5382 , 0x00},
+	{OV2685_8BIT, 0x5383 , 0x4a},
+	{OV2685_8BIT, 0x5384 , 0x00},
+	{OV2685_8BIT, 0x5385 , 0xb6},
+	{OV2685_8BIT, 0x5386 , 0x00},
+	{OV2685_8BIT, 0x5387 , 0x8d},
+	{OV2685_8BIT, 0x5388 , 0x00},
+	{OV2685_8BIT, 0x5389 , 0x3a},
+	{OV2685_8BIT, 0x538a , 0x00},
+	{OV2685_8BIT, 0x538b , 0xa6},
+	{OV2685_8BIT, 0x538c , 0x00},
+	{OV2685_8BIT, 0x5400 , 0x0d},
+	{OV2685_8BIT, 0x5401 , 0x18},
+	{OV2685_8BIT, 0x5402 , 0x31},
+	{OV2685_8BIT, 0x5403 , 0x5a},
+	{OV2685_8BIT, 0x5404 , 0x65},
+	{OV2685_8BIT, 0x5405 , 0x6f},
+	{OV2685_8BIT, 0x5406 , 0x77},
+	{OV2685_8BIT, 0x5407 , 0x80},
+	{OV2685_8BIT, 0x5408 , 0x87},
+	{OV2685_8BIT, 0x5409 , 0x8f},
+	{OV2685_8BIT, 0x540a , 0xa2},
+	{OV2685_8BIT, 0x540b , 0xb2},
+	{OV2685_8BIT, 0x540c , 0xcc},
+	{OV2685_8BIT, 0x540d , 0xe4},
+	{OV2685_8BIT, 0x540e , 0xf0},
+	{OV2685_8BIT, 0x540f , 0xa0},
+	{OV2685_8BIT, 0x5410 , 0x6e},
+	{OV2685_8BIT, 0x5411 , 0x06},
+	{OV2685_8BIT, 0x5480 , 0x19},
+	{OV2685_8BIT, 0x5481 , 0x00},
+	{OV2685_8BIT, 0x5482 , 0x09},
+	{OV2685_8BIT, 0x5483 , 0x12},
+	{OV2685_8BIT, 0x5484 , 0x04},
+	{OV2685_8BIT, 0x5485 , 0x06},
+	{OV2685_8BIT, 0x5486 , 0x08},
+	{OV2685_8BIT, 0x5487 , 0x0c},
+	{OV2685_8BIT, 0x5488 , 0x10},
+	{OV2685_8BIT, 0x5489 , 0x18},
+	{OV2685_8BIT, 0x5500 , 0x02},
+	{OV2685_8BIT, 0x5501 , 0x03},
+	{OV2685_8BIT, 0x5502 , 0x04},
+	{OV2685_8BIT, 0x5503 , 0x05},
+	{OV2685_8BIT, 0x5504 , 0x06},
+	{OV2685_8BIT, 0x5505 , 0x08},
+	{OV2685_8BIT, 0x5506 , 0x00},
+	{OV2685_8BIT, 0x5600 , 0x02},
+	{OV2685_8BIT, 0x5603 , 0x40},
+	{OV2685_8BIT, 0x5604 , 0x28},
+	{OV2685_8BIT, 0x5609 , 0x20},
+	{OV2685_8BIT, 0x560a , 0x60},
+	{OV2685_8BIT, 0x5800 , 0x03},
+	{OV2685_8BIT, 0x5801 , 0x24},
+	{OV2685_8BIT, 0x5802 , 0x02},
+	{OV2685_8BIT, 0x5803 , 0x40},
+	{OV2685_8BIT, 0x5804 , 0x34},
+	{OV2685_8BIT, 0x5805 , 0x05},
+	{OV2685_8BIT, 0x5806 , 0x12},
+	{OV2685_8BIT, 0x5807 , 0x05},
+	{OV2685_8BIT, 0x5808 , 0x03},
+	{OV2685_8BIT, 0x5809 , 0x3c},
+	{OV2685_8BIT, 0x580a , 0x02},
+	{OV2685_8BIT, 0x580b , 0x40},
+	{OV2685_8BIT, 0x580c , 0x26},
+	{OV2685_8BIT, 0x580d , 0x05},
+	{OV2685_8BIT, 0x580e , 0x52},
+	{OV2685_8BIT, 0x580f , 0x06},
+	{OV2685_8BIT, 0x5810 , 0x03},
+	{OV2685_8BIT, 0x5811 , 0x28},
+	{OV2685_8BIT, 0x5812 , 0x02},
+	{OV2685_8BIT, 0x5813 , 0x40},
+	{OV2685_8BIT, 0x5814 , 0x24},
+	{OV2685_8BIT, 0x5815 , 0x05},
+	{OV2685_8BIT, 0x5816 , 0x42},
+	{OV2685_8BIT, 0x5817 , 0x06},
+	{OV2685_8BIT, 0x5818 , 0x0d},
+	{OV2685_8BIT, 0x5819 , 0x40},
+	{OV2685_8BIT, 0x581a , 0x04},
+	{OV2685_8BIT, 0x581b , 0x0c},
+	{OV2685_8BIT, 0x3a03 , 0x4c},
+	{OV2685_8BIT, 0x3a04 , 0x40},
+	{OV2685_8BIT, 0x3503 , 0x00},
+	{OV2685_TOK_TERM, 0, 0}
+};
+
+/* resolution: 1280x720, ratio: 16:9, 1 lanes, 30fps */
+static struct ov2685_reg const ov2685_720p_init[] = {
+	/*1lane 30fps*/
+	{OV2685_8BIT, 0x0103 , 0x01},
+	{OV2685_8BIT, 0x3002 , 0x00},
+	{OV2685_8BIT, 0x3016 , 0x1c},
+	{OV2685_8BIT, 0x3018 , 0x44},
+	{OV2685_8BIT, 0x301d , 0xf0},
+	{OV2685_8BIT, 0x3020 , 0x00},
+	{OV2685_8BIT, 0x3082 , 0x37},
+	{OV2685_8BIT, 0x3083 , 0x03},
+	{OV2685_8BIT, 0x3084 , 0x0f},
+	{OV2685_8BIT, 0x3085 , 0x03},
+	{OV2685_8BIT, 0x3086 , 0x00},
+	{OV2685_8BIT, 0x3087 , 0x00},
+	{OV2685_8BIT, 0x3501 , 0x2d},
+	{OV2685_8BIT, 0x3502 , 0x80},
+	{OV2685_8BIT, 0x3503 , 0x03},
+	{OV2685_8BIT, 0x350b , 0x36},
+	{OV2685_8BIT, 0x3600 , 0xb4},
+	{OV2685_8BIT, 0x3603 , 0x35},
+	{OV2685_8BIT, 0x3604 , 0x24},
+	{OV2685_8BIT, 0x3605 , 0x00},
+	{OV2685_8BIT, 0x3620 , 0x26},
+	{OV2685_8BIT, 0x3621 , 0x37},
+	{OV2685_8BIT, 0x3622 , 0x04},
+	{OV2685_8BIT, 0x3628 , 0x10},
+	{OV2685_8BIT, 0x3705 , 0x3c},
+	{OV2685_8BIT, 0x370a , 0x21},
+	{OV2685_8BIT, 0x370c , 0x50},
+	{OV2685_8BIT, 0x370d , 0xc0},
+	{OV2685_8BIT, 0x3717 , 0x58},
+	{OV2685_8BIT, 0x3718 , 0x88},
+	{OV2685_8BIT, 0x3720 , 0x00},
+	{OV2685_8BIT, 0x3721 , 0x00},
+	{OV2685_8BIT, 0x3722 , 0x00},
+	{OV2685_8BIT, 0x3723 , 0x00},
+	{OV2685_8BIT, 0x3738 , 0x00},
+	{OV2685_8BIT, 0x3781 , 0x80},
+	{OV2685_8BIT, 0x3789 , 0x60},
+	/* output size settigs */
+	{OV2685_8BIT, 0x3800 , 0x00},/* xstart:160 */
+	{OV2685_8BIT, 0x3801 , 0xa0},
+	{OV2685_8BIT, 0x3802 , 0x00},/* ystart:242 */
+	{OV2685_8BIT, 0x3803 , 0xf2},
+	{OV2685_8BIT, 0x3804 , 0x05},/* xend: 1455 */
+	{OV2685_8BIT, 0x3805 , 0xaf},
+	{OV2685_8BIT, 0x3806 , 0x03},/* yend:973 */
+	{OV2685_8BIT, 0x3807 , 0xcd},
+	{OV2685_8BIT, 0x3808 , 0x05},/* xout:1280 */
+	{OV2685_8BIT, 0x3809 , 0x00},
+	{OV2685_8BIT, 0x380a , 0x02},/* yout:720 */
+	{OV2685_8BIT, 0x380b , 0xd0},
+	{OV2685_8BIT, 0x380c , 0x05},
+	{OV2685_8BIT, 0x380d , 0xa6},
+	{OV2685_8BIT, 0x380e , 0x02},
+	{OV2685_8BIT, 0x380f , 0xf8},
+	{OV2685_8BIT, 0x3810 , 0x00},
+	{OV2685_8BIT, 0x3811 , 0x08},
+	{OV2685_8BIT, 0x3812 , 0x00},
+	{OV2685_8BIT, 0x3813 , 0x06},
+	{OV2685_8BIT, 0x3814 , 0x11},
+	{OV2685_8BIT, 0x3815 , 0x11},
+	{OV2685_8BIT, 0x3819 , 0x04},
+	{OV2685_8BIT, 0x3820 , 0xc0},
+	{OV2685_8BIT, 0x3821 , 0x00},
+	/* AIQ settings */
+	{OV2685_8BIT, 0x3a06 , 0x00},
+	{OV2685_8BIT, 0x3a07 , 0xe4},
+	{OV2685_8BIT, 0x3a08 , 0x00},
+	{OV2685_8BIT, 0x3a09 , 0xbe},
+	{OV2685_8BIT, 0x3a0a , 0x15},
+	{OV2685_8BIT, 0x3a0b , 0x60},
+	{OV2685_8BIT, 0x3a0c , 0x17},
+	{OV2685_8BIT, 0x3a0d , 0xc0},
+	{OV2685_8BIT, 0x3a0e , 0x02},
+	{OV2685_8BIT, 0x3a0f , 0xac},
+	{OV2685_8BIT, 0x3a10 , 0x02},
+	{OV2685_8BIT, 0x3a11 , 0xf8},
+	{OV2685_8BIT, 0x4000 , 0x81},
+	{OV2685_8BIT, 0x4001 , 0x40},
+	{OV2685_8BIT, 0x4008 , 0x02},
+	{OV2685_8BIT, 0x4009 , 0x09},
+	{OV2685_8BIT, 0x4300 , 0x32},
+	{OV2685_8BIT, 0x430e , 0x00},
+	{OV2685_8BIT, 0x4602 , 0x02},
+	{OV2685_8BIT, 0x4837 , 0x1e},
+	{OV2685_8BIT, 0x5000 , 0xff},
+	{OV2685_8BIT, 0x5001 , 0x05},
+	{OV2685_8BIT, 0x5002 , 0x32},
+	{OV2685_8BIT, 0x5003 , 0x04},
+	{OV2685_8BIT, 0x5004 , 0xff},
+	{OV2685_8BIT, 0x5005 , 0x12},
+	{OV2685_8BIT, 0x0100 , 0x01},
+	{OV2685_8BIT, 0x5180 , 0xf4},
+	{OV2685_8BIT, 0x5181 , 0x11},
+	{OV2685_8BIT, 0x5182 , 0x41},
+	{OV2685_8BIT, 0x5183 , 0x42},
+	{OV2685_8BIT, 0x5184 , 0x78},
+	{OV2685_8BIT, 0x5185 , 0x58},
+	{OV2685_8BIT, 0x5186 , 0xb5},
+	{OV2685_8BIT, 0x5187 , 0xb2},
+	{OV2685_8BIT, 0x5188 , 0x08},
+	{OV2685_8BIT, 0x5189 , 0x0e},
+	{OV2685_8BIT, 0x518a , 0x0c},
+	{OV2685_8BIT, 0x518b , 0x4c},
+	{OV2685_8BIT, 0x518c , 0x38},
+	{OV2685_8BIT, 0x518d , 0xf8},
+	{OV2685_8BIT, 0x518e , 0x04},
+	{OV2685_8BIT, 0x518f , 0x7f},
+	{OV2685_8BIT, 0x5190 , 0x40},
+	{OV2685_8BIT, 0x5191 , 0x5f},
+	{OV2685_8BIT, 0x5192 , 0x40},
+	{OV2685_8BIT, 0x5193 , 0xff},
+	{OV2685_8BIT, 0x5194 , 0x40},
+	{OV2685_8BIT, 0x5195 , 0x07},
+	{OV2685_8BIT, 0x5196 , 0x04},
+	{OV2685_8BIT, 0x5197 , 0x04},
+	{OV2685_8BIT, 0x5198 , 0x00},
+	{OV2685_8BIT, 0x5199 , 0x05},
+	{OV2685_8BIT, 0x519a , 0xd2},
+	{OV2685_8BIT, 0x519b , 0x10},
+	{OV2685_8BIT, 0x5200 , 0x09},
+	{OV2685_8BIT, 0x5201 , 0x00},
+	{OV2685_8BIT, 0x5202 , 0x06},
+	{OV2685_8BIT, 0x5203 , 0x20},
+	{OV2685_8BIT, 0x5204 , 0x41},
+	{OV2685_8BIT, 0x5205 , 0x16},
+	{OV2685_8BIT, 0x5206 , 0x00},
+	{OV2685_8BIT, 0x5207 , 0x05},
+	{OV2685_8BIT, 0x520b , 0x30},
+	{OV2685_8BIT, 0x520c , 0x75},
+	{OV2685_8BIT, 0x520d , 0x00},
+	{OV2685_8BIT, 0x520e , 0x30},
+	{OV2685_8BIT, 0x520f , 0x75},
+	{OV2685_8BIT, 0x5210 , 0x00},
+	{OV2685_8BIT, 0x5280 , 0x14},
+	{OV2685_8BIT, 0x5281 , 0x02},
+	{OV2685_8BIT, 0x5282 , 0x02},
+	{OV2685_8BIT, 0x5283 , 0x04},
+	{OV2685_8BIT, 0x5284 , 0x06},
+	{OV2685_8BIT, 0x5285 , 0x08},
+	{OV2685_8BIT, 0x5286 , 0x0c},
+	{OV2685_8BIT, 0x5287 , 0x10},
+	{OV2685_8BIT, 0x5300 , 0xc5},
+	{OV2685_8BIT, 0x5301 , 0xa0},
+	{OV2685_8BIT, 0x5302 , 0x06},
+	{OV2685_8BIT, 0x5303 , 0x0a},
+	{OV2685_8BIT, 0x5304 , 0x30},
+	{OV2685_8BIT, 0x5305 , 0x60},
+	{OV2685_8BIT, 0x5306 , 0x90},
+	{OV2685_8BIT, 0x5307 , 0xc0},
+	{OV2685_8BIT, 0x5308 , 0x82},
+	{OV2685_8BIT, 0x5309 , 0x00},
+	{OV2685_8BIT, 0x530a , 0x26},
+	{OV2685_8BIT, 0x530b , 0x02},
+	{OV2685_8BIT, 0x530c , 0x02},
+	{OV2685_8BIT, 0x530d , 0x00},
+	{OV2685_8BIT, 0x530e , 0x0c},
+	{OV2685_8BIT, 0x530f , 0x14},
+	{OV2685_8BIT, 0x5310 , 0x1a},
+	{OV2685_8BIT, 0x5311 , 0x20},
+	{OV2685_8BIT, 0x5312 , 0x80},
+	{OV2685_8BIT, 0x5313 , 0x4b},
+	{OV2685_8BIT, 0x5380 , 0x01},
+	{OV2685_8BIT, 0x5381 , 0x52},
+	{OV2685_8BIT, 0x5382 , 0x00},
+	{OV2685_8BIT, 0x5383 , 0x4a},
+	{OV2685_8BIT, 0x5384 , 0x00},
+	{OV2685_8BIT, 0x5385 , 0xb6},
+	{OV2685_8BIT, 0x5386 , 0x00},
+	{OV2685_8BIT, 0x5387 , 0x8d},
+	{OV2685_8BIT, 0x5388 , 0x00},
+	{OV2685_8BIT, 0x5389 , 0x3a},
+	{OV2685_8BIT, 0x538a , 0x00},
+	{OV2685_8BIT, 0x538b , 0xa6},
+	{OV2685_8BIT, 0x538c , 0x00},
+	{OV2685_8BIT, 0x5400 , 0x0d},
+	{OV2685_8BIT, 0x5401 , 0x18},
+	{OV2685_8BIT, 0x5402 , 0x31},
+	{OV2685_8BIT, 0x5403 , 0x5a},
+	{OV2685_8BIT, 0x5404 , 0x65},
+	{OV2685_8BIT, 0x5405 , 0x6f},
+	{OV2685_8BIT, 0x5406 , 0x77},
+	{OV2685_8BIT, 0x5407 , 0x80},
+	{OV2685_8BIT, 0x5408 , 0x87},
+	{OV2685_8BIT, 0x5409 , 0x8f},
+	{OV2685_8BIT, 0x540a , 0xa2},
+	{OV2685_8BIT, 0x540b , 0xb2},
+	{OV2685_8BIT, 0x540c , 0xcc},
+	{OV2685_8BIT, 0x540d , 0xe4},
+	{OV2685_8BIT, 0x540e , 0xf0},
+	{OV2685_8BIT, 0x540f , 0xa0},
+	{OV2685_8BIT, 0x5410 , 0x6e},
+	{OV2685_8BIT, 0x5411 , 0x06},
+	{OV2685_8BIT, 0x5480 , 0x19},
+	{OV2685_8BIT, 0x5481 , 0x00},
+	{OV2685_8BIT, 0x5482 , 0x09},
+	{OV2685_8BIT, 0x5483 , 0x12},
+	{OV2685_8BIT, 0x5484 , 0x04},
+	{OV2685_8BIT, 0x5485 , 0x06},
+	{OV2685_8BIT, 0x5486 , 0x08},
+	{OV2685_8BIT, 0x5487 , 0x0c},
+	{OV2685_8BIT, 0x5488 , 0x10},
+	{OV2685_8BIT, 0x5489 , 0x18},
+	{OV2685_8BIT, 0x5500 , 0x02},
+	{OV2685_8BIT, 0x5501 , 0x03},
+	{OV2685_8BIT, 0x5502 , 0x04},
+	{OV2685_8BIT, 0x5503 , 0x05},
+	{OV2685_8BIT, 0x5504 , 0x06},
+	{OV2685_8BIT, 0x5505 , 0x08},
+	{OV2685_8BIT, 0x5506 , 0x00},
+	{OV2685_8BIT, 0x5600 , 0x02},
+	{OV2685_8BIT, 0x5603 , 0x40},
+	{OV2685_8BIT, 0x5604 , 0x28},
+	{OV2685_8BIT, 0x5609 , 0x20},
+	{OV2685_8BIT, 0x560a , 0x60},
+	{OV2685_8BIT, 0x5800 , 0x03},
+	{OV2685_8BIT, 0x5801 , 0x24},
+	{OV2685_8BIT, 0x5802 , 0x02},
+	{OV2685_8BIT, 0x5803 , 0x40},
+	{OV2685_8BIT, 0x5804 , 0x34},
+	{OV2685_8BIT, 0x5805 , 0x05},
+	{OV2685_8BIT, 0x5806 , 0x12},
+	{OV2685_8BIT, 0x5807 , 0x05},
+	{OV2685_8BIT, 0x5808 , 0x03},
+	{OV2685_8BIT, 0x5809 , 0x3c},
+	{OV2685_8BIT, 0x580a , 0x02},
+	{OV2685_8BIT, 0x580b , 0x40},
+	{OV2685_8BIT, 0x580c , 0x26},
+	{OV2685_8BIT, 0x580d , 0x05},
+	{OV2685_8BIT, 0x580e , 0x52},
+	{OV2685_8BIT, 0x580f , 0x06},
+	{OV2685_8BIT, 0x5810 , 0x03},
+	{OV2685_8BIT, 0x5811 , 0x28},
+	{OV2685_8BIT, 0x5812 , 0x02},
+	{OV2685_8BIT, 0x5813 , 0x40},
+	{OV2685_8BIT, 0x5814 , 0x24},
+	{OV2685_8BIT, 0x5815 , 0x05},
+	{OV2685_8BIT, 0x5816 , 0x42},
+	{OV2685_8BIT, 0x5817 , 0x06},
+	{OV2685_8BIT, 0x5818 , 0x0d},
+	{OV2685_8BIT, 0x5819 , 0x40},
+	{OV2685_8BIT, 0x581a , 0x04},
+	{OV2685_8BIT, 0x581b , 0x0c},
+	{OV2685_8BIT, 0x3a03 , 0x4c},
+	{OV2685_8BIT, 0x3a04 , 0x40},
+	{OV2685_8BIT, 0x3503 , 0x00},
+	{OV2685_TOK_TERM, 0, 0}
+};
+
+/* resolution: 800 x 600, ratio: 4:3, 2 lanes, 30fps */
+static struct ov2685_reg const ov2685_svga_init[] = {
+	/*2lanes, 30fps*/
+	{OV2685_8BIT, 0x0103 , 0x01},
+	{OV2685_8BIT, 0x3002 , 0x00},
+	{OV2685_8BIT, 0x3016 , 0x1c},
+	{OV2685_8BIT, 0x3018 , 0x84},
+	{OV2685_8BIT, 0x301d , 0xf0},
+	{OV2685_8BIT, 0x3020 , 0x00},
+	{OV2685_8BIT, 0x3082 , 0x37},/*mclk = 19.2Mhz*/
+	{OV2685_8BIT, 0x3083 , 0x03},
+	{OV2685_8BIT, 0x3084 , 0x07},
+	{OV2685_8BIT, 0x3085 , 0x03},
+	{OV2685_8BIT, 0x3086 , 0x00},
+	{OV2685_8BIT, 0x3087 , 0x00},
+	{OV2685_8BIT, 0x3501 , 0x26},
+	{OV2685_8BIT, 0x3502 , 0x40},
+	{OV2685_8BIT, 0x3503 , 0x03},
+	{OV2685_8BIT, 0x350b , 0x36},
+	{OV2685_8BIT, 0x3600 , 0xb4},
+	{OV2685_8BIT, 0x3603 , 0x35},
+	{OV2685_8BIT, 0x3604 , 0x24},
+	{OV2685_8BIT, 0x3605 , 0x00},
+	{OV2685_8BIT, 0x3620 , 0x24},
+	{OV2685_8BIT, 0x3621 , 0x34},
+	{OV2685_8BIT, 0x3622 , 0x03},
+	{OV2685_8BIT, 0x3628 , 0x10},
+	{OV2685_8BIT, 0x3705 , 0x3c},
+	{OV2685_8BIT, 0x370a , 0x21},
+	{OV2685_8BIT, 0x370c , 0x50},
+	{OV2685_8BIT, 0x370d , 0xc0},
+	{OV2685_8BIT, 0x3717 , 0x58},
+	{OV2685_8BIT, 0x3718 , 0x80},
+	{OV2685_8BIT, 0x3720 , 0x00},
+	{OV2685_8BIT, 0x3721 , 0x09},
+	{OV2685_8BIT, 0x3722 , 0x0b},
+	{OV2685_8BIT, 0x3723 , 0x48},
+	{OV2685_8BIT, 0x3738 , 0x99},
+	{OV2685_8BIT, 0x3781 , 0x80},
+	{OV2685_8BIT, 0x3784 , 0x0c},
+	{OV2685_8BIT, 0x3789 , 0x60},
+	/* output size settings */
+	{OV2685_8BIT, 0x3800 , 0x00},/* xstart:0 */
+	{OV2685_8BIT, 0x3801 , 0x00},
+	{OV2685_8BIT, 0x3802 , 0x00},/* ystart:0 */
+	{OV2685_8BIT, 0x3803 , 0x00},
+	{OV2685_8BIT, 0x3804 , 0x06},/* xend:1615 */
+	{OV2685_8BIT, 0x3805 , 0x4f},
+	{OV2685_8BIT, 0x3806 , 0x04},/* yend:1215 */
+	{OV2685_8BIT, 0x3807 , 0xbf},
+	{OV2685_8BIT, 0x3808 , 0x03},/* xout:800 */
+	{OV2685_8BIT, 0x3809 , 0x20},
+	{OV2685_8BIT, 0x380a , 0x02},/* yout:600 */
+	{OV2685_8BIT, 0x380b , 0x58},
+	{OV2685_8BIT, 0x380c , 0x06},
+	{OV2685_8BIT, 0x380d , 0xa4},
+	{OV2685_8BIT, 0x380e , 0x05},
+	{OV2685_8BIT, 0x380f , 0x0e},
+	{OV2685_8BIT, 0x3810 , 0x00},
+	{OV2685_8BIT, 0x3811 , 0x04},
+	{OV2685_8BIT, 0x3812 , 0x00},
+	{OV2685_8BIT, 0x3813 , 0x04},
+	{OV2685_8BIT, 0x3814 , 0x31},
+	{OV2685_8BIT, 0x3815 , 0x31},
+	{OV2685_8BIT, 0x3819 , 0x04},
+	{OV2685_8BIT, 0x3820 , 0xc2},
+	{OV2685_8BIT, 0x3821 , 0x01},
+	/* AIQ settings */
+	{OV2685_8BIT, 0x3a06 , 0x01},
+	{OV2685_8BIT, 0x3a07 , 0x84},
+	{OV2685_8BIT, 0x3a08 , 0x01},
+	{OV2685_8BIT, 0x3a09 , 0x43},
+	{OV2685_8BIT, 0x3a0a , 0x24},
+	{OV2685_8BIT, 0x3a0b , 0x60},
+	{OV2685_8BIT, 0x3a0c , 0x28},
+	{OV2685_8BIT, 0x3a0d , 0x60},
+	{OV2685_8BIT, 0x3a0e , 0x04},
+	{OV2685_8BIT, 0x3a0f , 0x8c},
+	{OV2685_8BIT, 0x3a10 , 0x05},
+	{OV2685_8BIT, 0x3a11 , 0x0c},
+	{OV2685_8BIT, 0x3a13 , 0x80},
+	{OV2685_8BIT, 0x4000 , 0x81},
+	{OV2685_8BIT, 0x4001 , 0x40},
+	{OV2685_8BIT, 0x4008 , 0x00},
+	{OV2685_8BIT, 0x4009 , 0x03},
+	{OV2685_8BIT, 0x4300 , 0x32},
+	{OV2685_8BIT, 0x430e , 0x00},
+	{OV2685_8BIT, 0x4602 , 0x02},
+	{OV2685_8BIT, 0x4837 , 0x1e},
+	{OV2685_8BIT, 0x5000 , 0xff},
+	{OV2685_8BIT, 0x5001 , 0x05},
+	{OV2685_8BIT, 0x5002 , 0x32},
+	{OV2685_8BIT, 0x5003 , 0x0c},
+	{OV2685_8BIT, 0x5004 , 0xff},
+	{OV2685_8BIT, 0x5005 , 0x12},
+	{OV2685_8BIT, 0x0100 , 0x01},
+	{OV2685_8BIT, 0x5180 , 0xf4},
+	{OV2685_8BIT, 0x5181 , 0x11},
+	{OV2685_8BIT, 0x5182 , 0x41},
+	{OV2685_8BIT, 0x5183 , 0x42},
+	{OV2685_8BIT, 0x5184 , 0x78},
+	{OV2685_8BIT, 0x5185 , 0x58},
+	{OV2685_8BIT, 0x5186 , 0xb5},
+	{OV2685_8BIT, 0x5187 , 0xb2},
+	{OV2685_8BIT, 0x5188 , 0x08},
+	{OV2685_8BIT, 0x5189 , 0x0e},
+	{OV2685_8BIT, 0x518a , 0x0c},
+	{OV2685_8BIT, 0x518b , 0x4c},
+	{OV2685_8BIT, 0x518c , 0x38},
+	{OV2685_8BIT, 0x518d , 0xf8},
+	{OV2685_8BIT, 0x518e , 0x04},
+	{OV2685_8BIT, 0x518f , 0x7f},
+	{OV2685_8BIT, 0x5190 , 0x40},
+	{OV2685_8BIT, 0x5191 , 0x5f},
+	{OV2685_8BIT, 0x5192 , 0x40},
+	{OV2685_8BIT, 0x5193 , 0xff},
+	{OV2685_8BIT, 0x5194 , 0x40},
+	{OV2685_8BIT, 0x5195 , 0x07},
+	{OV2685_8BIT, 0x5196 , 0x04},
+	{OV2685_8BIT, 0x5197 , 0x04},
+	{OV2685_8BIT, 0x5198 , 0x00},
+	{OV2685_8BIT, 0x5199 , 0x05},
+	{OV2685_8BIT, 0x519a , 0xd2},
+	{OV2685_8BIT, 0x519b , 0x10},
+	{OV2685_8BIT, 0x5200 , 0x09},
+	{OV2685_8BIT, 0x5201 , 0x00},
+	{OV2685_8BIT, 0x5202 , 0x06},
+	{OV2685_8BIT, 0x5203 , 0x20},
+	{OV2685_8BIT, 0x5204 , 0x41},
+	{OV2685_8BIT, 0x5205 , 0x16},
+	{OV2685_8BIT, 0x5206 , 0x00},
+	{OV2685_8BIT, 0x5207 , 0x05},
+	{OV2685_8BIT, 0x520b , 0x30},
+	{OV2685_8BIT, 0x520c , 0x75},
+	{OV2685_8BIT, 0x520d , 0x00},
+	{OV2685_8BIT, 0x520e , 0x30},
+	{OV2685_8BIT, 0x520f , 0x75},
+	{OV2685_8BIT, 0x5210 , 0x00},
+	{OV2685_8BIT, 0x5280 , 0x14},
+	{OV2685_8BIT, 0x5281 , 0x02},
+	{OV2685_8BIT, 0x5282 , 0x02},
+	{OV2685_8BIT, 0x5283 , 0x04},
+	{OV2685_8BIT, 0x5284 , 0x06},
+	{OV2685_8BIT, 0x5285 , 0x08},
+	{OV2685_8BIT, 0x5286 , 0x0c},
+	{OV2685_8BIT, 0x5287 , 0x10},
+	{OV2685_8BIT, 0x5300 , 0xc5},
+	{OV2685_8BIT, 0x5301 , 0xa0},
+	{OV2685_8BIT, 0x5302 , 0x06},
+	{OV2685_8BIT, 0x5303 , 0x0a},
+	{OV2685_8BIT, 0x5304 , 0x30},
+	{OV2685_8BIT, 0x5305 , 0x60},
+	{OV2685_8BIT, 0x5306 , 0x90},
+	{OV2685_8BIT, 0x5307 , 0xc0},
+	{OV2685_8BIT, 0x5308 , 0x82},
+	{OV2685_8BIT, 0x5309 , 0x00},
+	{OV2685_8BIT, 0x530a , 0x26},
+	{OV2685_8BIT, 0x530b , 0x02},
+	{OV2685_8BIT, 0x530c , 0x02},
+	{OV2685_8BIT, 0x530d , 0x00},
+	{OV2685_8BIT, 0x530e , 0x0c},
+	{OV2685_8BIT, 0x530f , 0x14},
+	{OV2685_8BIT, 0x5310 , 0x1a},
+	{OV2685_8BIT, 0x5311 , 0x20},
+	{OV2685_8BIT, 0x5312 , 0x80},
+	{OV2685_8BIT, 0x5313 , 0x4b},
+	{OV2685_8BIT, 0x5380 , 0x01},
+	{OV2685_8BIT, 0x5381 , 0x52},
+	{OV2685_8BIT, 0x5382 , 0x00},
+	{OV2685_8BIT, 0x5383 , 0x4a},
+	{OV2685_8BIT, 0x5384 , 0x00},
+	{OV2685_8BIT, 0x5385 , 0xb6},
+	{OV2685_8BIT, 0x5386 , 0x00},
+	{OV2685_8BIT, 0x5387 , 0x8d},
+	{OV2685_8BIT, 0x5388 , 0x00},
+	{OV2685_8BIT, 0x5389 , 0x3a},
+	{OV2685_8BIT, 0x538a , 0x00},
+	{OV2685_8BIT, 0x538b , 0xa6},
+	{OV2685_8BIT, 0x538c , 0x00},
+	{OV2685_8BIT, 0x5400 , 0x0d},
+	{OV2685_8BIT, 0x5401 , 0x18},
+	{OV2685_8BIT, 0x5402 , 0x31},
+	{OV2685_8BIT, 0x5403 , 0x5a},
+	{OV2685_8BIT, 0x5404 , 0x65},
+	{OV2685_8BIT, 0x5405 , 0x6f},
+	{OV2685_8BIT, 0x5406 , 0x77},
+	{OV2685_8BIT, 0x5407 , 0x80},
+	{OV2685_8BIT, 0x5408 , 0x87},
+	{OV2685_8BIT, 0x5409 , 0x8f},
+	{OV2685_8BIT, 0x540a , 0xa2},
+	{OV2685_8BIT, 0x540b , 0xb2},
+	{OV2685_8BIT, 0x540c , 0xcc},
+	{OV2685_8BIT, 0x540d , 0xe4},
+	{OV2685_8BIT, 0x540e , 0xf0},
+	{OV2685_8BIT, 0x540f , 0xa0},
+	{OV2685_8BIT, 0x5410 , 0x6e},
+	{OV2685_8BIT, 0x5411 , 0x06},
+	{OV2685_8BIT, 0x5480 , 0x19},
+	{OV2685_8BIT, 0x5481 , 0x00},
+	{OV2685_8BIT, 0x5482 , 0x09},
+	{OV2685_8BIT, 0x5483 , 0x12},
+	{OV2685_8BIT, 0x5484 , 0x04},
+	{OV2685_8BIT, 0x5485 , 0x06},
+	{OV2685_8BIT, 0x5486 , 0x08},
+	{OV2685_8BIT, 0x5487 , 0x0c},
+	{OV2685_8BIT, 0x5488 , 0x10},
+	{OV2685_8BIT, 0x5489 , 0x18},
+	{OV2685_8BIT, 0x5500 , 0x02},
+	{OV2685_8BIT, 0x5501 , 0x03},
+	{OV2685_8BIT, 0x5502 , 0x04},
+	{OV2685_8BIT, 0x5503 , 0x05},
+	{OV2685_8BIT, 0x5504 , 0x06},
+	{OV2685_8BIT, 0x5505 , 0x08},
+	{OV2685_8BIT, 0x5506 , 0x00},
+	{OV2685_8BIT, 0x5600 , 0x02},
+	{OV2685_8BIT, 0x5603 , 0x40},
+	{OV2685_8BIT, 0x5604 , 0x28},
+	{OV2685_8BIT, 0x5609 , 0x20},
+	{OV2685_8BIT, 0x560a , 0x60},
+	{OV2685_8BIT, 0x5800 , 0x03},
+	{OV2685_8BIT, 0x5801 , 0x24},
+	{OV2685_8BIT, 0x5802 , 0x02},
+	{OV2685_8BIT, 0x5803 , 0x40},
+	{OV2685_8BIT, 0x5804 , 0x34},
+	{OV2685_8BIT, 0x5805 , 0x05},
+	{OV2685_8BIT, 0x5806 , 0x12},
+	{OV2685_8BIT, 0x5807 , 0x05},
+	{OV2685_8BIT, 0x5808 , 0x03},
+	{OV2685_8BIT, 0x5809 , 0x3c},
+	{OV2685_8BIT, 0x580a , 0x02},
+	{OV2685_8BIT, 0x580b , 0x40},
+	{OV2685_8BIT, 0x580c , 0x26},
+	{OV2685_8BIT, 0x580d , 0x05},
+	{OV2685_8BIT, 0x580e , 0x52},
+	{OV2685_8BIT, 0x580f , 0x06},
+	{OV2685_8BIT, 0x5810 , 0x03},
+	{OV2685_8BIT, 0x5811 , 0x28},
+	{OV2685_8BIT, 0x5812 , 0x02},
+	{OV2685_8BIT, 0x5813 , 0x40},
+	{OV2685_8BIT, 0x5814 , 0x24},
+	{OV2685_8BIT, 0x5815 , 0x05},
+	{OV2685_8BIT, 0x5816 , 0x42},
+	{OV2685_8BIT, 0x5817 , 0x06},
+	{OV2685_8BIT, 0x5818 , 0x0d},
+	{OV2685_8BIT, 0x5819 , 0x40},
+	{OV2685_8BIT, 0x581a , 0x04},
+	{OV2685_8BIT, 0x581b , 0x0c},
+	{OV2685_8BIT, 0x3a03 , 0x4c},
+	{OV2685_8BIT, 0x3a04 , 0x40},
+	{OV2685_8BIT, 0x3503 , 0x00},
+	{OV2685_TOK_TERM, 0, 0}
+};
 static struct ov2685_reg const ov2685_common[] = {
 	 {OV2685_TOK_TERM, 0, 0}
 };
-- 
1.9.1

